### DISCUSSION TITLE
FDS visualisation with ParaView

### DISCUSSION THREAD

**Discussion by LordNarcisse** [firemodels/fds#12822] [Q&A]
Hi everybody !

I got a problem by reading .q and .xyz files in ParaView. Every timesteps and variables are in a totally disorder on ParaView.
I'm here to understand how to modify the .q and .xyz file on FDS to make them readable on ParaView.

Here's my code : 
[Test.txt](https://github.com/firemodels/fds/files/15023154/Test.txt)

Also, it exist a .smv reader on ParaView but it seems not to be quite ready yet to read these file types.

Has anyone ever had the same type of problem?

Thank you,

AN

**Comment by rmcdermo** [comment]
Hi.  I don't not have any experience trying to use the PLOT3D output for Paraview.  We have decided it makes more sense to develop VTK output, which is native to Paraview.  This is available in a research branch of FDS called FireX, now available here:

https://github.com/marcosvanella/fds/tree/FireX

You can output VTK using WRITE_VTK=T on DUMP.

@johodges can point us to instructions beyond this.

**Reply by LordNarcisse** [reply]
Thank you for your answer. Is the generation of VTK files planned for a future official version of FDS or is it only for the FireX branch ? I'm not sure to know how to use FireX but I will do my best !

**Reply by rmcdermo** [reply]
Planned for future FDS release.  But this will be months away probably.

**Reply by johodges** [reply]
The current vtk branch outside of FireX is at the fork below. You should not need to do anything special to build this fork other than checking out the branch. FDS generates a paraview statefile which you can use to read in the data (.py file) when WRITE_VTK is true. Let me know if you run into issues and I'll look into it. Planning to put together a more detailed wiki sometime over the next month.

https://github.com/johodges/fds/tree/vtk-out

**Reply by LordNarcisse** [reply]
Thank you. However I don't really understand how github forks work. I downloaded the .zip folder containing fds-vtk-out but I cannot run fds. Do I need to download something else ?

**Reply by rmcdermo** [reply]
You would need to [compile the code](https://github.com/firemodels/fds/wiki/FDS-Compilation).  If this is not something you are comfortable with, then you will need to wait awhile until we push up a test release with this feature.  As I said, it will be months away.

**Reply by LordNarcisse** [reply]
Ok perfect! I will try to complicate it and I will keep you informed. Thanks again

**Reply by Muthu-V** [reply]
Any update on this  ? I want to visualize FDS field data in paraview.

**Reply by johodges** [reply]
The vtk output currently lives on a fork (https://github.com/johodges/fds/tree/vtkhdf) so you will need to compile FDS yourself to use it. There are instructions on the wiki (https://github.com/firemodels/fds/wiki/Working-with-VTK-Files). The output is functional right now but the file footprint is large as the geometry is output at every time step. We are making some changes to address that over the next couple of weeks.

**Reply by Muthu-V** [reply]
Thanks for letting me know. I will wait for this functionality.  I hope, Particles can also be exported with the field values .

**Reply by johodges** [reply]
Could you clarify what you would like exported with the particles? The current implementation exports all of the PARTICLE_QUANTITIES linked to each PART class.

**Reply by Muthu-V** [reply]
That is good enough .
